embd.go

```diff
@@ -72,20 +73,36 @@ func run() error {
 		Args:  os.Args[1:],
 		Pkg:   *pkg,
 		Files: map[string]File{},
+		Dirs:  make(map[string][]File),
 	}
 	for _, path := range flag.Args() {
-		// TODO(akavel): support directories
-		f, err := NewFile(path)
+		fi, err := os.Stat(path)
 		if err != nil {
 			return err
 		}
-		if old, exists := contents.Files[f.VarName]; exists {
-			return fmt.Errorf(
-				"generated variable name conflict: '%s' resolves to"+
-					"the same variable name %s as '%s'",
-				f.Path, f.VarName, old.Path)
+
+		if fi.IsDir() {
+			f, _ := ioutil.ReadDir(path)
```

Comment:
1. Rename 'f' to 'dir'.
2. Please don't ignore error (something could have happened since Stat, also this is a different operation so it may fail because it needs deeper access.)